.github/workflows/containers-ghcr.yml

```diff
@@ -0,0 +1,79 @@
+name: "Build and Publish Release Containers to GHCR"
+
+on:
+  release:
+    types: [published]
+
+env:
+  REGISTRY: ghcr.io
+
+permissions: read-all
+
+jobs:
+  build_and_publish:
+    name: "Build and Publish Release Containers"
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        platform: [sgx, virtual, snp]
+        type: [dev, run]
+        run_js: [true, ""]
+        clang_version: ["11", "15"]
+        exclude:
+          - type: dev
+            run_js: true
+          - platform: sgx
+            clang_version: "15"
+          - platform: virtual
+            clang_version: "11"
+          - platform: snp
+            clang_version: "11"
```

Comment:
Doing the platform <-> clang_version pairing with exclusions seems risky, can't we bind these together in the matrix?

```yaml
matrix:
  platform:
    - name: sgx
      clang_version: 11
    - name: virtual
      clang_version: 15
    - name: snp
      clang_version: 15
```

Reply:
I'm inclined to keep it because 1. it works, and 2. it will disappear with SGX.